src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -263,28 +286,82 @@ overwrite_ops_t prepare_ops_list(
       assert(to_write.size() > 1);
       assert(back.addr + back.len ==
 	back.pin->get_key() + back.pin->get_length());
-      ops.to_remap.push_back(extent_to_remap_t::create_remap(
+      ops.to_remap.push_back(extent_to_remap_t::create_remap1(
 	std::move(back.pin),
 	back.addr - back.pin->get_key(),
 	back.len));
       ops.to_remove.pop_back();
     }
   }
 
-  // prepare to_insert
+  interval_set<uint64_t> pre_alloc_addr_removed, pre_alloc_addr_remapped;
+  if (delta_based_overwrite_max_extent_size) {
+    for (auto &r : ops.to_remove) {
+      if (r->get_val().is_absolute()) {
+	pre_alloc_addr_removed.insert(r->get_key(), r->get_length());
+
+      }
+    }
+    for (auto &r : ops.to_remap) {
+      if (r.pin && r.pin->get_val().is_absolute()) {
```

Comment:
Same to the line 300 above.

The meaning of `pin->get_val().is_absolute()` is dependent based on different seastore backends (Segment/RBM).

IIUC this PR is generic among backends, so is it more approprite to check whether the extent referred by the mapping is stable?

Reply:
It seems like a right way to go. I'll update this after is_stable() is introduced.

Reply:
In addition to is_stable(), we need to add ` is_real() `to filter out zero data. If not, `read_extent()` fails. What do you think?